test/e2e/components/vl-http-error-message.js

```diff
@@ -1,51 +1,58 @@
 const { VlElement } = require('vl-ui-core');
 const { VlTypography } = require('vl-ui-typography');
+const { VlButton } = require('vl-ui-button');
 const { By } = require('selenium-webdriver');
 
 class VlHttpErrorMessage extends VlElement {
-    async getImageSource() {
-        const smallImage = await this.shadowRoot.findElement(By.css('#image-small'));
-        const normalImage = await this.shadowRoot.findElement(By.css('#image-normal'));
-        const smallImageSrc = await smallImage.getAttribute("src");
-        const normalImageSrc = await normalImage.getAttribute("src");
-        if (smallImageSrc != normalImageSrc) {
-        	throw new Error("Small en normal image src moeten gelijk zijn");
-        }
-        return smallImageSrc;
+    async getTitle() {
+        return (await this._getTitle()).getText();
     }
 
-    async getImageAlt() {
-    	const smallImage = await this.shadowRoot.findElement(By.css('#image-small'));
-    	const normalImage = await this.shadowRoot.findElement(By.css('#image-normal'));
-    	const smallImageAlt = await smallImage.getAttribute("alt");
-    	const normalImageAlt = await normalImage.getAttribute("alt");
-        if (smallImageAlt != normalImageAlt) {
-        	throw new Error("Small en normal image src moeten gelijk zijn");
-        }
-    	return smallImageAlt;
+    async getContent() {
+        return (await this._getTypography()).getText();
     }
 
-    async getTitle() {
-    	const title = await this.shadowRoot.findElement(By.css('#title'));
-    	return title.getText();
+    async clickOnAction() {
+        return (await this._getAction()).click();
     }
 
-    async getLinkText() {
-    	const link = await this.shadowRoot.findElement(By.css('a'));
-    	return link.getText();
+    async _getTitle() {
+        return this.shadowRoot.findElement(By.css('#title'));
     }
 
-    async getContent() {
-    	const textElement = await this.shadowRoot.findElement(By.css('#text'));
-    	const typography = await new VlTypography(this.driver, textElement);
-        return typography.getInnerHTML();
+    async _getTypography() {
+        const element = await this.shadowRoot.findElement(By.css('#text'));
+        return new VlTypography(this.driver, element);
     }
 
-    async clickOnLink() {
-    	const link = await this.shadowRoot.findElement(By.css('a'));
-        return link.click();
+    async _getAction() {
+        const element = await this.shadowRoot.findElement(By.css('a'));
+        return new VlButton(this.driver, element);
```

Comment:
nen button voor nen a?

Reply:
Geïntroduceerd door @SanderKleykens. Er is een button met de looks van een link en omgekeerd. Doordat we geen cyclic dependencies willen, hebben we besloten de link die er uit ziet als een button in de vl-ui-button te declareren. Hierdoor moet je dus het button page object gebruiken.

``` JavaScript
define('vl-link-button', VlLinkButton, {extends: 'a'});
```

Reply:
brrrr